### PR TITLE
Use the term content instead of payload

### LINKED ISSUE #654: Multiple kinds of payload in HTTP/2 and HTTP/3 conflict with payload terminology from core and RFC7230
Taking another look at the problems surrounding the term "payload", I went to see what was being used in h2 (RFC7540) and HTTP/3. Unfortunately, they use both "payload body" (h2, 7230) / "payload data" (h3, http-core) and also "frame payload" (h2, h3).

So, at various times, DATA payload != payload data, but a DATA payload might contain the payload data if it happens to be the only DATA frame.

Just shoot me.

### PR DESCRIPTION
**Pull request by royfielding**
Fixes #654

and makes the whole spec read a lot easier because now the major feature names align with the content of a message.

### CODE REVIEW

**Comment by royfielding** [comment]
Note that there is one remaining instance of payload (in the PUT section) that I have not changed yet because it would create a merge conflict with #653

**Review by mnot — COMMENTED** [review]
This looks much better.

I still am not 100% clear on the relationship between a representation and content; perhaps I need to see things in situ.

If a representation consists of a header section, content, and one or more trailer sections (with the associated status code if it's a response, method and url if it's a request), that makes sense to me, and we should come out and say it explicitly. However, there are a few places in the edits below which indicate that equates content with a representation, so I think they need to be adjusted (e.g., 'representation in this message's content').

For me, that brings up the question of what the concrete difference is now between the message abstraction (defined in semantics 6) and representations (defined in semantics 8). I strongly suspect they can be collapsed into one set of terms. Failing that, they should at least be aligned so the relationships (and differences, if any) are clear.

File: draft-ietf-httpbis-messaging-latest.xml [hunk 1]
```diff
@@ -327,7 +327,7 @@
 </t>
 <t>
    A sender &MUST-NOT; generate a bare CR (a CR character not immediately
-   followed by LF) within any protocol elements other than the payload data.
+   followed by LF) within any protocol elements other than the content.
```
[the remark this answers is not in the document]
Comment (royfielding) [reply draft-ietf-httpbis-messaging-latest.xml:330]: I prefer it as just the content. The way I am thinking is that the content is the stream of octets that would be extracted from the framing of a message regardless of HTTP version.

[the remark this answers is not in the document]
Comment (royfielding) [reply draft-ietf-httpbis-messaging-latest.xml:330]: E.g., in h2 and h3, the DATA frames each have a payload from which data octets are extracted in order to form the request/response content.

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -2063,10 +2062,10 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
    Note that this message abstraction is a generalization across many versions
    of HTTP, including features that might not be found in some versions. For
    example, trailers were introduced within the HTTP/1.1 chunked transfer
-   coding as a single trailer section at the end of the payload data. An
+   coding as a single trailer section at the end of the content. An
```
[the remark this answers is not in the document]
Comment (royfielding) [reply draft-ietf-httpbis-semantics-latest.xml:2065]: wouldn't that be "at the end of the content" or "after the content"? Either one is fine with me. "after the end" seems weird.

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -2180,86 +2179,85 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
...
-<section title="Payload" anchor="payload">
-<iref item="payload"/>
+<section title="Content" anchor="content">
+<iref item="content"/>
 <t>
    HTTP messages often transfer a complete or partial representation as the
-   message <x:dfn>payload</x:dfn>, including both representation metadata
-   transferred as fields and representation data transferred as
-   <x:dfn>payload data</x:dfn>: a stream of octets sent after the header
+   message <x:dfn>content</x:dfn>: a stream of octets sent after the header
```
[the remark this answers is not in the document]
Comment (royfielding) [reply draft-ietf-httpbis-semantics-latest.xml:2186]: Right.

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml:2065]:
```suggestion
   coding as a single trailer section after the content. An
```

File: draft-ietf-httpbis-messaging-latest.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnot) [reply draft-ietf-httpbis-messaging-latest.xml:330]: Makes sense, but see the overall comment -> https://github.com/httpwg/http-core/pull/655#pullrequestreview-563179664

**Comment by royfielding** [comment]
> For me, that brings up the question of what the concrete difference is now between the message abstraction (defined in semantics 6) and representations (defined in semantics 8). I strongly suspect they can be collapsed into one set of terms. Failing that, they should at least be aligned so the relationships (and differences, if any) are clear.

The content of a message is what gets slung across the wire. It is usually the representation data of something, but often isn't present in a message (even when the selected representation contains data). This actually matches very well with the meaning of the Content-* fields, assuming we refer to those fields seperately from the content.

A representation is a value for a resource, including metadata and data, more like a database record that has multiple fields.

We want to talk about them separately because most of semantics is defining how to apply methods with content to a resource function, resulting in a state change to the target resource. Some methods change the selected representation metadata without sending a representation. Some methods change the selected representation data without sending metadata. Talking about them as if they were all messages with content doesn't work in practice, aside from within caches.

**Review by reschke — APPROVED** [review]

**Comment by mnot** [comment]
I was still reviewing this @reschke

**Comment by royfielding** [comment]
sorry, my fault -- I thought I was just waiting for Julian

**Pull request merged**